components/tools/OmeroPy/test/integration/clitest/test_pyramids.py

```diff
@@ -0,0 +1,235 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+
+from test.integration.clitest.cli import CLITest
+from omero.cli import NonZeroReturnCode
+from omero.model import PixelsI
+from omero.rtypes import rint
+from omero.rtypes import unwrap
+from omero.sys import ParametersI
+
+import os
+import omero.plugins.admin
+import pytest
+import time
+import datetime
+
+
+class TestRemovePyramids(CLITest):
+
+    def setup_method(self, method):
+        super(TestRemovePyramids, self).setup_method(method)
+        self.cli.register("admin", omero.plugins.admin.AdminControl, "TEST")
+        self.args += ["admin", "removepyramids"]
+
+    def test_removepyramids_admin_only(self, capsys):
+        """Test removepyramids is admin-only"""
+        with pytest.raises(NonZeroReturnCode):
+            self.cli.invoke(self.args, strict=True)
+        out, err = capsys.readouterr()
+        assert err.endswith("SecurityViolation: Admins only!\n")
+
+
+class TestRemovePyramidsRestrictedAdmin(CLITest):
+
+    # make the user in this test a member of system group
+    DEFAULT_SYSTEM = True
+    # make the new member of system group to a Restricted
+    # Admin with no privileges
+    DEFAULT_PRIVILEGES = ()
+
+    def setup_method(self, method):
+        super(TestRemovePyramidsRestrictedAdmin, self).setup_method(method)
+        self.cli.register("admin", omero.plugins.admin.AdminControl, "TEST")
+        self.args += ["admin", "removepyramids"]
+
+    def test_removepyramids_restricted_admin(self, capsys):
+        """Test removepyramids cannot be run by Restricted Admin"""
+        with pytest.raises(NonZeroReturnCode):
+            self.cli.invoke(self.args, strict=True)
+        out, err = capsys.readouterr()
+        output_end = "SecurityViolation: Admin restrictions:"
+        assert err.startswith(output_end)
+
+
+class TestRemovePyramidsFullAdmin(CLITest):
+
+    # make the user in this test a member of system group
+    DEFAULT_SYSTEM = True
+    # make the new member of system group to a Full Admin
+    DEFAULT_PRIVILEGES = None
+
+    def setup_method(self, method):
+        super(TestRemovePyramidsFullAdmin, self).setup_method(method)
+        self.cli.register("admin", omero.plugins.admin.AdminControl, "TEST")
+        self.args += ["admin", "removepyramids"]
+        self.group_ctx = {'omero.group': str(self.group.id.val)}
+
+    def import_pyramid(self, tmpdir, name=None):
+        if name is None:
+            name = "test&sizeX=4000&sizeY=4000.fake"
+        fakefile = tmpdir.join(name)
+        fakefile.write('')
+        pixels = self.import_image(filename=str(fakefile), skip="checksum")[0]
+        # wait for the pyramid to be generated
+        id = long(float(pixels))
+        assert id >= 0
+        time.sleep(30)
```

Comment:
`time.sleep` should be used only as a last resort. Would be good to replace these with a `for`-loop which checks to see if the pyramids have been generated yet. (A `omero.cmd` callback would be even better)

Reply:
Agree. We can improve that in follow-up PR